libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1018,6 +1017,22 @@ void SMTEncoder::visitFunctionIdentifier(Identifier const& _identifier)
 	}
 }
 
+void SMTEncoder::visitStructConstructorCall(FunctionCall const& _funCall) {
+	solAssert(*_funCall.annotation().kind == FunctionCallKind::StructConstructorCall, "");
+	auto structSymbolicVar = dynamic_pointer_cast<smt::SymbolicStructVariable>(m_context.expression(_funCall));
+	solAssert(structSymbolicVar, "");
+	TypePointer t = _funCall.annotation().type;
+	auto structType = dynamic_cast<StructType const*>(t);
+	solAssert(structType, "");
+
+	// for each member of the struct, set the corresponding argument in the constructor as the field of the struct
+	auto members = structType->nativeMembers(nullptr);
+	auto const sortedArguments = _funCall.sortedArguments();
+	solAssert(sortedArguments.size() == members.size(), "");
+	for (unsigned i = 0; i < members.size(); ++i)
+		m_context.addAssertion(expr(*sortedArguments[i]) == structSymbolicVar->member(members[i].name));
```

Comment:
This looks fine, but I think it could make sense to create a new method in `SymbolicStructVariable` that takes the sorted arguments, increases its own SSA index, and asserts the above

Reply:
That's a good idea, I will try that.

Reply:
Do we need to increase the index? It should be a constructor, so there should be no previous value associated with it.

Reply:
Yea, I just think it might be safer. I don't know if, for example, `createExpr` is called on the FunctionCall expression first which creates a zero-assigned object, in which case the assertions would be unsat. I think it's just safer to increase the index, and it doesn't harm